resources/js/validation.js

```diff
@@ -0,0 +1,188 @@
+    if (value === '') {
+      showError(nicknameInput, ErrorMessage.nicknameRequired);
+      return false;
+    }
+    return true;
+  }
+  
+  // 비밀번호 확인 유효성 검사 (회원가입 페이지 전용)
+  function checkConfirmPassword() {
+    if (!isSignupPage) return true;
+    const confirmValue = confirmPasswordInput.value.trim();
+    const passwordValue = passwordInput.value.trim();
+    removeError(confirmPasswordInput);
+    if (confirmValue === '') {
+      showError(confirmPasswordInput, ErrorMessage.confirmPasswordRequired);
+      return false;
+    }
+    if (confirmValue !== passwordValue) {
+      showError(confirmPasswordInput, ErrorMessage.passwordMismatch);
+      return false;
+    }
+    return true;
+  }
+  
+  // 버튼 상태 업데이트 함수
+  function updateButtonState() {
+    const hasError = form.querySelector('.error-message') !== null;
+    let allFieldsFilled =
+      emailInput.value.trim() !== '' &&
+      passwordInput.value.trim() !== '' &&
+      passwordInput.value.trim().length >= 8;
+    
+    if (isSignupPage) {
+      allFieldsFilled =
+        allFieldsFilled &&
+        nicknameInput.value.trim() !== '' &&
+        confirmPasswordInput.value.trim() !== '';
+    }
+    
+    if (!hasError && allFieldsFilled) {
+      submitButton.disabled = false;
+      submitButton.title = isLoginPage ? '로그인 가능상태' : '회원가입 가능상태';
+      submitButton.textContent = isLoginPage ? '로그인' : '회원가입';
+    } else {
+      submitButton.disabled = true;
+      submitButton.title = isLoginPage ? '정확한 로그인 정보를 입력하세요' : '정확한 회원가입 정보를 입력하세요';
+      submitButton.textContent = isLoginPage ? '로그인' : '회원가입';
+    }
+  }
+  
+  // 이벤트 리스너 등록
+  emailInput.addEventListener('blur', checkEmail);
+  emailInput.addEventListener('input', updateButtonState);
+  
+  passwordInput.addEventListener('blur', checkPassword);
+  passwordInput.addEventListener('input', () => {
+    if (isSignupPage && confirmPasswordInput.value.trim() !== '') {
+      checkConfirmPassword();
+    }
+    updateButtonState();
+  });
+  
+  if (isSignupPage) {
+    nicknameInput.addEventListener('blur', checkNickname);
+    nicknameInput.addEventListener('input', updateButtonState);
+    
+    confirmPasswordInput.addEventListener('blur', checkConfirmPassword);
+    confirmPasswordInput.addEventListener('input', updateButtonState);
+  }
+  
+  form.addEventListener('submit', function(event) {
+    event.preventDefault();
+    if (isLoginPage) {
+      const isEmailValid = checkEmail();
+      const isPasswordValid = checkPassword();
+      if (isEmailValid && isPasswordValid) {
+        navigateTo('../../html/items.html');
+      }
+    } else if (isSignupPage) {
+      const isEmailValid = checkEmail();
+      const isNicknameValid = checkNickname();
+      const isPasswordValid = checkPassword();
+      const isConfirmPasswordValid = checkConfirmPassword();
+      if (isEmailValid && isNicknameValid && isPasswordValid && isConfirmPasswordValid) {
+        navigateTo('../../html/login.html');
+      }
+    }
```

Comment:
💊 제안
이런 경우 early return 문을 쓰시면 좋을 것 같아요!

```suggestion
    if (isLoginPage) {
      const isEmailValid = checkEmail();
      const isPasswordValid = checkPassword();
      
      if (isEmailValid && isPasswordValid) {
        navigateTo('../../html/items.html');
      }
      
      return;
    } 
    
    const isEmailValid = checkEmail();
    const isNicknameValid = checkNickname();
    const isPasswordValid = checkPassword();
    const isConfirmPasswordValid = checkConfirmPassword();
    if (isEmailValid && isNicknameValid && isPasswordValid && isConfirmPasswordValid) {
      navigateTo('../../html/login.html');
    }
```